shc/web/interface.py

```diff
@@ -291,11 +321,34 @@ async def _page_handler(self, request: aiohttp.web.Request) -> aiohttp.web.Respo
 
         html_title = self.title_formatter(page.title)
         template = jinja_env.get_template('page.htm')
-        body = await template.render_async(title=page.title, segments=page.segments, menu=self.ui_menu_entries,
-                                           root_url=self.root_url, js_files=self._js_files, css_files=self._css_files,
-                                           server_token=id(self), html_title=html_title)
+        self._mark_active_menu_items(page.name)
+        body = await template.render_async(
+            title=page.title,
+            segments=page.segments,
+            menu=self.ui_menu_entries,
+            root_url=self.root_url,
+            js_files=self._js_files,
+            css_files=self._css_files,
+            server_token=id(self),
+            html_title=html_title,
+            SubMenuEntrySpec=SubMenuEntrySpec,
+        )
         return aiohttp.web.Response(body=body, content_type="text/html", charset='utf-8')
 
+    def _mark_active_menu_items(self, page_name: str):
+        """Set menu items is_active flag if current page matches page_name/target link."""
+        for item in self.ui_menu_entries:
+            if isinstance(item, SubMenuEntrySpec):
+                item.is_active = False
+                for sub_item in item.submenus:
+                    if page_name == sub_item.page_name:
+                        item.is_active = True
+                        sub_item.is_active = True
+                    else:
+                        sub_item.is_active = False
+            else:
+                item.is_active = page_name == item.page_name
```

Comment:
I really don't like that this method changes the `is_active` attribute of the MenuEntrySpecs in the (server-global) `self.ui_menu_entries` attribute, such that this is now a side-effect of the `_page_handler()` method.

This may even result in concurrency issues, considering, that `template.render_async()` may be executed concurrently for multiple concurrent requests. So, the `is_active` value should definitely only be changed in a local data structure within the `_page_handler()` method.

Reply:
Ok, understood & fixed.